src/averageAges.js

```diff
@@ -55,7 +67,17 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const difference = (!onlyWithSon)
+    ? people.filter(child =>
+      people.find(mother => child.mother === mother.name))
+    : people.filter(child =>
+      people.find(mother => child.mother === mother.name) && child.sex === 'm');
```

Comment:
[checklist point 2](https://github.com/mate-academy/js_average-ages/blob/master/checklist.md)

In this case better use `some` then `find` method.

It would be clearly to create for lines 72 and 74 variables `hasMother` for example.